### DISCUSSION TITLE
The appshell header doesn't occupy the entire width.

### DISCUSSION THREAD

**Discussion by imjishnu001** [skeletonlabs/skeleton#1829] [Q&A] [answered]
Hi Team, 

I have been struggling to get the Appshell header and body content to cover the complete width of the screen. 

I have update the global css and app.html as suggested in the docs

`html,
body {
	@apply h-full overflow-hidden;
}`

`<bodydata-sveltekit-preload-data="hover" data-theme="skeleton">
		<div style="display: contents"  class="h-full overflow-hidden">%sveltekit.body%</div>
</body>`

however this only covers the height, I tried manually changing the width but it only works in "px". I believe I am missing something really basic, any suggestions on how to fix the issue?

Thanks in advance!

<img width="1440" alt="image" src="https://github.com/skeletonlabs/skeleton/assets/66176194/6b62f216-3411-4d56-94fe-c646a46f4853">

**Comment by Sarenor** [comment]
Can you share your repository? It might be that you have a margin applied around the AppShell. The Skeleton Docs themselves are build with the AppShell and it covers the full width there

**Reply by imjishnu001** [reply]
Thank you for the prompt response.

I haven't pushed the project yet. 
below is the code for the global and main layout pages if it helps. 

`<script lang='ts'>
	// The ordering of these imports is critical to your app working properly
	//import '../theme.postcss';
	import '@skeletonlabs/skeleton/themes/theme-skeleton.css';
	// If you have source.organizeImports set to true in VSCode, then it will auto change this ordering
	import '@skeletonlabs/skeleton/styles/skeleton.css';
	// Most of your app wide CSS should be put in this file
	import '../app.postcss';
	import { Toast } from '@skeletonlabs/skeleton';
	
</script>
	<link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css" rel="stylesheet"> 

	<div class="container h-full">
		<slot />
	</div>
`



`<script lang='ts'>
	
	import { Avatar, AppBar, AppShell, Drawer, drawerStore, Toast, Modal } from '@skeletonlabs/skeleton';
	import Navigation from '$lib/components/Navigation.svelte';
	import { LightSwitch } from '@skeletonlabs/skeleton';


	function drawerOpen(): void{
		drawerStore.open()
	}
</script>

<Toast position="br"/>
<Modal />
<Drawer>
	<Navigation />
</Drawer>

<AppShell slotSidebarLeft="w-0 md:w-60 bg-surface-500/10">
	<svelte:fragment slot="header">
		<AppBar>
			<svelte:fragment slot="lead">
				<button class="md:hidden btn btn-sm mr-4" on:click={drawerOpen}>
					<span>
						<svg viewBox="0 0 100 80" class="fill-token w-4 h4">
							<rect width="100" height="20"/>
							<rect y="30" width="100" height="20"/>
							<rect y="60" width="100" height="20"/>
						</svg>
					</span>
				</button>
				<span class="fa-solid fa-handshake text-6xl"></span>
				<strong class="text-xl uppercase">ProjectDDP</strong>
				
				
			</svelte:fragment>
			<svelte:fragment slot="trail">
					<LightSwitch />
			</svelte:fragment>
		</AppBar>
	</svelte:fragment>
	<svelte:fragment slot="sidebarLeft">
		<Navigation />
	</svelte:fragment>
	
	
	<div class="container p-10 w-full h-full">
		<slot />
	</div>
	
</AppShell>
`

**Reply by Sarenor** [reply]
Why the separation in global and main layout? 
If the AppShell resides in main and the ``container h-full`` is the global one, that's probably where the extra margin comes from.

**Comment by endigo9740** [comment]
@imjishnu001 assuming this isn't a typo, there is a missing space here between the words body and data:

```
<bodydata-sveltekit-preload-data="hover" data-theme="skeleton">
```

I'd start there, but otherwise I'd recommend right clicking and inspecting your markup in your browser and try to find where the space is coming from. We've not seen this for other projects, so it's something on your end.

If you can provide a minimal repro we can take a look as well

**Reply by imjishnu001** [reply]
Thank you, 
That was indeed a typo in the body tag. I am working on getting the code pushed to Git Hub, just need to make a few changes. 

Below are screenshots from the developer console. I haven't added any custom CSS. 

![image](https://github.com/skeletonlabs/skeleton/assets/66176194/4ff66a30-b7ae-4225-89f0-378f28fd2800)

![image](https://github.com/skeletonlabs/skeleton/assets/66176194/236410ed-38f8-4ef7-b460-6feda6f7c3d5)

**Comment by imjishnu001** [comment, marked as answer]
The issue stems from an unfortunate design choice I made.

Appshell consumes full width when it is in the global layouts page, whereas I have added it to layout of a route, possibly this is causing issue.

**Reply by Sarenor** [reply]
That is probably the case, yeah.
